### PR TITLE
Add `no-dupe-class-members` rule with support for class properties

### LINKED ISSUE #190: Add no-dupe-class-members rule that supports ClassProperty
see https://github.com/eslint/eslint/issues/13059

```js
class A { foo = 42; get foo() {} }
```

If the rule will not be adjusted in the ESLint repo we should consider copying the rule here and adjusting it to work for Stage 3 class fields too.

### PR DESCRIPTION
**Pull request by Turbo87**
This PR continues #173, but adds a bit of code to detect and ignore the common `foo = this.foo.bind(this);` pattern, that was mentioned in #190.

Resolves #190

/cc @ljharb

### CODE REVIEW

File: tests/rules/no-dupe-class-members.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+          type: "ClassProperty",
+          line: 1,
+          column: 20,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        }
+      ]
+    },
+    {
+      code: "class A { foo = 1; foo = 1; foo = 1; }",
```
Comment (ljharb) [inline tests/rules/no-dupe-class-members.js]:
```suggestion
      code: "class A { foo = 1; foo = 2; foo = 3; }",
```
so we make sure to check different values too

File: tests/rules/no-dupe-class-members.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+          type: "ClassProperty",
+          line: 1,
+          column: 29,
+          messageId: "unexpected",
+          data: { name: "foo" }
+        }
+      ]
+    },
+    {
+      code: "class A { static foo = 1; static foo = 1; }",
```
Comment (ljharb) [inline tests/rules/no-dupe-class-members.js]:
```suggestion
      code: "class A { static foo = 1; static foo = 2; }",
```

**Review by ljharb — COMMENTED** [review]
Can we include an invalid case of `foo() {}` and `foo = ` something that's not `this.foo.bind(this)`?

**Comment by Turbo87** [comment]
@ljharb done, thanks for the quick review

**Comment by Turbo87** [comment]
something about CI appears to be strange. opening TravisCI shows CI as "passed", but at GitHub it still shows as pending. not sure what to do about this 🤔

**Comment by existentialism** [comment]
@Turbo87 really our fault for not making this more obvious, but we're consolidating most of our eslint-related repos into the [main monorepo](https://github.com/babel/babel/issues/10752).

In anticipation of this, mind opening a PR for it over there? I can assist if you're too busy, sorry for the trouble.

**Comment by Turbo87** [comment]
I'll have a look. thanks for the warning @existentialism :)

**Comment by kaicataldo** [comment]
Thank you for the PR. Now that [@babel/eslint-plugin has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).

**Pull request closed without merging**